hikari/internal/compatibility.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Nekokatt
+# Copyright (c) 2021 davfsa
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""A module that provides components to provide backwards compatibility."""
+
+from __future__ import annotations
+
+__all__: typing.List[str] = ["Pattern"]
+
+import re
+import sys
+import typing
+
+CURRENT_PYTHON_VERSION = sys.version_info
+
+# Backwards compatibility for `re.Pattern`
+#
+# Explanation
+# -----------
+# Before Python 3.9, you had to use `typing.Pattern` to mark a regex pattern.
+# This was later moved to the `re` module under `re.Pattern` and deprecated
+# `typing.Pattern`.
+#
+# Provides backwards compatibility for: Python 3.8.0 - 3.9.0
+
+if CURRENT_PYTHON_VERSION <= (3, 9, 0):
+    Pattern = typing.Pattern
+else:
+    Pattern = re.Pattern
```

Comment:
Do we even need to do this when the typing stuff hasn't been removed or should we also love stuff like typing.(Mutable)Mapping, typing.Dict, typing.Type, typing.(Mutable)Seqience, typing.(Abstract)Set, etc etc here as well

Reply:
If it's marked in the python documentation as `Deprecated as of {version}` and what it has been deprecated for is not in all python versions, then yes

Reply:
Not at anytime, they specify a timeline for it and even mention it in the docs and this change covers half of the typing module it's not worth the effort to change all of them to this when by the time it's removed well have removed support for 3.8

Reply:
Ah, ok then :+1:

Reply:
Most of the time when stuff is deprecated there's a timeline given for when it'll be removed, that's just the same way to do stuff